src/sbvr-api/translations.ts

```diff
@@ -0,0 +1,216 @@
+		| Dictionary<string | ReferencedFieldNode>
+	> = {},
+): Dictionary<string> => {
+	const isDefinition = (d: typeof definitions[string]): d is Definition =>
+		'abstractSql' in d;
+
+	const resourceRenames: Dictionary<string> = {};
+
+	// TODO: why?
+	fromAbstractSqlModel.rules = toAbstractSqlModel.rules;
+
+	const fromKeys = Object.keys(fromAbstractSqlModel.tables);
+	const nonexistentTables = _.difference(Object.keys(definitions), fromKeys);
+	if (nonexistentTables.length > 0) {
+		throw new Error(
+			`Tried to define non-existent resources: '${nonexistentTables.join(
+				', ',
+			)}'`,
+		);
+	}
+	_.forEach(toAbstractSqlModel.synonyms, (canonicalForm, synonym) => {
+		// Don't double alias
+		if (synonym.includes('$')) {
+			fromAbstractSqlModel.synonyms[synonym] = canonicalForm;
+		} else {
+			fromAbstractSqlModel.synonyms[
+				`${synonym}$${toVersion}`
+			] = `${canonicalForm}$${toVersion}`;
+		}
+	});
+	const relationships = _.cloneDeep(toAbstractSqlModel.relationships);
+	namespaceRelationships(relationships, toVersion);
+	_.forEach(relationships, (relationship, key) => {
+		// Don't double alias
+		if (!key.includes('$')) {
+			key = `${key}$${toVersion}`;
+		}
+		fromAbstractSqlModel.relationships[key] = relationship;
+	});
+
+	// TODO: We also need to keep the original relationship refs to non $version resources
+	// Also alias for ourselves to allow explicit referencing
+	const aliasedFromRelationships = _.cloneDeep(
+		fromAbstractSqlModel.relationships,
+	);
+	namespaceRelationships(aliasedFromRelationships as Relationship, fromVersion);
+	_.forEach(aliasedFromRelationships, (relationship, key) => {
+		// Don't double alias
+		if (!key.includes('$')) {
+			key = `${key}$${fromVersion}`;
+			fromAbstractSqlModel.relationships[key] = relationship;
+		}
+	});
+
+	_.forEach(toAbstractSqlModel.tables, (table, key) => {
+		// Don't double alias
+		if (!key.includes('$')) {
+			key = `${key}$${toVersion}`;
+		}
+		fromAbstractSqlModel.tables[key] = _.cloneDeep(table);
+	});
+
+	for (const key of fromKeys) {
+		const definition = definitions[key];
+		const table = fromAbstractSqlModel.tables[key];
+		if (definition) {
+			const { $toResource } = definition;
+			const hasToResource = typeof $toResource === 'string';
+			if (hasToResource) {
+				resourceRenames[key] = `${$toResource}`;
+			}
+			const toResource = hasToResource ? $toResource : `${key}$${toVersion}`;
+			// TODO: Should this use the toAbstractSqlModel?
+			const toTable = fromAbstractSqlModel.tables[toResource];
+			if (!toTable) {
+				if (hasToResource) {
+					throw new Error(`Unknown $toResource: '${toResource}'`);
+				} else {
+					throw new Error(`Missing $toResource: '${toResource}'`);
+				}
+			}
+			table.modifyFields = _.cloneDeep(toTable.modifyFields ?? toTable.fields);
+			table.modifyName = toTable.modifyName ?? toTable.name;
+			if (isDefinition(definition)) {
+				const d = { ...definition };
+				delete d.$toResource;
```

Comment:
``` suggestion
				const d: Definition = { ...definition };
				delete d.$toResource;
```

`$toResource` can be undefined and `delete d.$toResource;` would fail, I guess this is okay for typescript as typeof `d` evaluates to `any` after the spread operator expand

Reply:
`delete d.$toResource;` won't fail if it's undefined, it'll just be a no-op

Reply:
Oh yeah, I tried it in node and it's just a no op returning `true`.